python-bindings/pythonbindings.cpp

```diff
@@ -139,8 +143,8 @@ PYBIND11_MODULE(blspy, m)
             [](const G1Element &pk,
                const py::bytes &msg,
                const G2Element &sig) {
+                py::gil_scoped_release release;
                 std::string s(msg);
```

Comment:
I think you're releasing the GIL too early here. `msg` is a python type, I don't think you can read from it after releasing the GIL.